tools/eubfr-cli/bin/cli.js

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env node
+
+const program = require('commander');
+const readline = require('readline');
+const pkg = require('../package.json');
+
+// Utilities
+const ensureVariables = require('../lib/ensureVariables');
+const getCredentials = require('../lib/getProducerCredentials');
+const getEndpoints = require('../lib/getEndpoints');
+const getAllProducers = require('../lib/getAllProducers');
+
+// Commands
+const uploadFiles = require('../commands/upload');
+const showFile = require('../commands/show');
+const deleteFiles = require('../commands/delete');
+
+// If -p is passed without actual value, it will be boolean true
+// Which is useless information in our case
+const hasValidProducer = options =>
+  options.producer && typeof options.producer !== 'boolean';
+
+const endpoints = getEndpoints();
+const missingRequiredInput = '\n error: Missing required input parameters';
+
+program.version(pkg.version).usage('[command] [option]');
+
+program
+  .command('upload [files...]')
+  .description('Uploads content to the data lake.')
+  .option('-p, --producer [producer]', "Producer's name.")
+  .action((files, options) => {
+    ensureVariables(['SIGNED_UPLOADS_API'], endpoints);
+
+    let credentials = [];
+    const producerIsSet = hasValidProducer(options);
+    const filesAreSet = files.length !== 0;
+
+    if (!producerIsSet && filesAreSet) {
+      console.error(missingRequiredInput);
+      process.exit(1);
+    }
+
+    if (producerIsSet) {
+      const { producer } = options;
+      credentials = [{ [producer]: getCredentials(producer) }];
+    }
+
+    if (!producerIsSet && !filesAreSet) {
+      console.log('All files for all producers will be uploaded.');
+      const producers = getAllProducers();
+      credentials = producers.map(producer => ({
+        [producer]: getCredentials(producer),
+      }));
+    }
+
+    uploadFiles({ files, credentials, endpoints });
```

Comment:
I maintain it would be better to `await` this instead of just letting the Promise resolve in the background 😉